xc/xc7/tests/soc/litex/base/CMakeLists.txt

```diff
@@ -1,44 +1,52 @@
-add_file_target(FILE mem.init)
-add_file_target(FILE mem_1.init)
-add_file_target(FILE mem_2.init)
-add_file_target(FILE arty.sdc)
 add_file_target(FILE arty_clocks.xdc)
-add_file_target(FILE arty.pcf)
-add_file_target(FILE baselitex_arty.xdc)
 
-add_file_target(FILE VexRiscv_Linux.v SCANNER_TYPE verilog)
-add_file_target(FILE baselitex_arty.v SCANNER_TYPE verilog)
-add_fpga_target(
-  NAME baselitex_arty
-  BOARD arty-full
-  SOURCES
-    baselitex_arty.v
-    VexRiscv_Linux.v
-  INPUT_IO_FILE arty.pcf
-  INPUT_XDC_FILE baselitex_arty.xdc
-  EXPLICIT_ADD_FILE_TARGET
-)
+function(litex_test litex_dir board device test_name)
+  set(LITEX ${CMAKE_CURRENT_BINARY_DIR}/${litex_dir}/gateware)
 
-add_vivado_target(
-    NAME baselitex_arty_vivado
-    PARENT_NAME baselitex_arty
-    XDC arty_clocks.xdc
-)
+  get_file_location(GENERATE_LITEX_LOC ${GENERATE_LITEX})
+  get_file_location(FIXUP_XDC_LOC ${FIXUP_XDC})
+  add_custom_command(
+    OUTPUT
+      ${LITEX}/arty.v
+      ${LITEX}/arty.xdc
+      ${LITEX}/arty.pcf
+      ${LITEX}/arty.sdc
+      ${LITEX}/mem.init
+      ${LITEX}/mem_1.init
+      ${LITEX}/mem_2.init
+    DEPENDS ${PYTHON3} ${GENERATE_LITEX_LOC} ${FIXUP_XDC_LOC}
```

Comment:
Use `append_file_dependency` for creating dependency lists.

Reply:
Done